File: javascript/vowel.js
// Implement the function which should return true if given object is a vowel (meaning a, e, i, o, u, uppercase or lowercase), and false otherwise.

String.prototype.vowel = function() {
  return /^[aeiou]$/i.test(this)
};

describe('vowel', () => {
  it('Basic tests', () => {
    assert.strictEqual(''.vowel(), false);
    assert.strictEqual('a'.vowel(), true);
    assert.strictEqual('E'.vowel(), true);
    assert.strictEqual('ou'.vowel(), false);
    assert.strictEqual('z'.vowel(), false);
    assert.strictEqual('lol'.vowel(), false);
  });
});

File: python/vowel.py
# https://www.codewars.com/kata/567bed99ee3451292c000025

# Implement the function which should return true if given object is a vowel (meaning a, e, i, o, u, uppercase or lowercase), and false otherwise.

import re

def is_vowel(s): 
    return bool(re.fullmatch("^[aeiou]$", s, flags = re.I))
    
@test.describe('Fixed Tests')
def fixed_tests():

    @test.it('Basic Test Cases')
    def basic_tests():
        test.assert_equals(is_vowel(""), False)
        test.assert_equals(is_vowel("a"), True)
        test.assert_equals(is_vowel("E"), True)
        test.assert_equals(is_vowel("ou"), False)
        test.assert_equals(is_vowel("z"), False)
        test.assert_equals(is_vowel("lol"), False)
